admin/class-wp-security-bp-admin.php

```diff
@@ -169,23 +185,83 @@ public function add_action_links( $links ) {
 	}
 
 	/**
-	 * This function calls all functions and agroup all return in to one json.
+	 * This function calls all checking methods and returns a JSON.
 	 *
 	 * @since    1.0.0
 	 */
-	public function final_json(){
+	public function check_all(){
+
 		//Class Files calls:
 		$files = new WP_Security_BP_Files( $this->plugin_name, $this->admin_url );
-		$json_return[] = $files->check_wp_config();
+		$response[] = $files->check_wp_config();
+
 		//Class Users calls:
 		$users = new WP_Security_BP_Users( $this->plugin_name, $this->admin_url );
-		$json_return[] = $users->check_users_ids();
+		$response[] = $users->check_users_ids();
+
     	//Class Database checks:
 		$db = new WP_Security_BP_Database( $this->plugin_name );
-		$json_return[] = $db->check_name();
+		$response[] = $db->check_name();
+		
+		wp_send_json( $response );
+
+	}
+
+	/**
+	 * This function gets the action sent through POST and calls the appropiate method.
+	 *
+	 * @since    1.0.0
+	 */
+	public function run_ajax_calls() {
+
+		if ( wp_doing_ajax() ) {
+			
+			$action = empty( $_POST['action'] ) ? '' : $_POST['action'];
+			
+			if ( $action === 'check-all' ) {
+				$this->check_all();
+			}
+			
+			/**
+			 * This array matches the 'action' value sent with the ajax request with the
+			 * corresponding method that should be fired.
+			 * 
+			 * The key is the action passed from JS. All action names must include the exact 
+			 * name of the class that should be fired followed by a hyphen '-' and some identitiy name.
+			 * 
+			 * The value is the method name and must be exact excluding the brackets '()'.
+			 */
+			$actions = array(
+				'class-example-action' => 'example_class_method', // for example purpose only
+				'files-fix-wp-config' => 'fix_wp_config',
+			);
+
+			if ( array_key_exists( $action, $actions ) ) {
+
+				$key = strstr( $action, '-', true );
+				$method = $actions[$action];
+
+				switch ( $key ) {
+					case 'files':
+						$class = new WP_Security_BP_Files( $this->plugin_name, $this->admin_url );
+						break;
+					case 'users':
+						$class = new WP_Security_BP_Users( $this->plugin_name );
+						break;
+					case 'database':
+						$class = new WP_Security_BP_Database( $this->plugin_name );
+						break;
+					default:
+						wp_die();
+				}
+				
+				$class->$method();
+				$this->check_all();
+				
+			}
+
+		}
```

Comment:
There is a Lack of alternative. what to do if $action is not in $actions. It's in every check we do in this class almost.

In fact, to be fair, there is a lack of alternatives at plugin wide level.(what to do if not the expected parameter or iteration)

we must start refactor this.


There is nothing to do with this part of the code if variable doesn't match the actual cardcoded array. OK. But we don't pay attention at next. I mean , the rest of the code stills there and logic happens in one way or another. 

As this is an open source development, we must take even more care in this details. 

🔐

Reply:
Totally agree 👍